src/ty/mod.rs

```diff
@@ -411,6 +415,50 @@ where
     }
 }
 
+/// Type describing a [`Range`].
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(
+    feature = "serde",
+    serde(bound(
+        serialize = "T::Type: Serialize, T::String: Serialize",
+        deserialize = "T::Type: DeserializeOwned, T::String: DeserializeOwned",
+    ))
+)]
+#[cfg_attr(any(feature = "std", feature = "decode"), derive(scale::Decode))]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Encode, Debug)]
+pub struct TypeDefRange<T: Form = MetaForm> {
+    start: T::Type,
+    end: T::Type,
+    inclusive: bool,
+}
+
+impl TypeDefRange {
+    /// Creates a new [`TypeDefRange`] for the supplied index type. Use the `inclusive` parameter to
+    /// distinguish between [`Range`] and [`RangeInclusive`] range types.
```

Comment:
I don't think these links will resolve because `Range` and `RangeInclusive` are not in scope here?

Reply:
good pt!